internal/engine/ingester/diff/diff.go

```diff
@@ -203,6 +208,124 @@ func (di *Diff) ingestFileForDepDiff(
 	return batchCtxDeps, nil
 }
 
+func (di *Diff) getScalibrTypeDiff(ctx context.Context, _ int, pr *pb.PullRequest) (*engif.Result, error) {
+	deps := pbinternal.PrDependencies{Pr: pr}
+
+	// TODO: we should be able to just fetch the additional commits between base and target.
+	// Our current Git abstraction isn't quite powerful enough, so we do two shallow clones.
+
+	baseInventory, err := di.scalibrInventory(ctx, pr.BaseCloneUrl, pr.BaseRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone base from %s at %q: %w", pr.BaseCloneUrl, pr.BaseRef, err)
+	}
+	newInventory, err := di.scalibrInventory(ctx, pr.TargetCloneUrl, pr.TargetRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone fork from %s at %q: %w", pr.TargetCloneUrl, pr.TargetRef, err)
+	}
+
+	newDeps := setDifference(baseInventory, newInventory, inventorySorter)
+
+	deps.Deps = make([]*pbinternal.PrDependencies_ContextualDependency, 0, len(newDeps))
+	for _, inventory := range newDeps {
+		for _, filename := range inventory.Locations {
+			deps.Deps = append(deps.Deps, &pbinternal.PrDependencies_ContextualDependency{
+				Dep: &pbinternal.Dependency{
+					Ecosystem: inventoryToEcosystem(inventory),
+					Name:      inventory.Name,
+					Version:   inventory.Version,
+				},
+				File: &pbinternal.PrDependencies_ContextualDependency_FilePatch{
+					Name:     filename,
+					PatchUrl: "", // TODO: do we need this?
+				},
+			})
+		}
+	}
+
+	return &engif.Result{Object: &deps, Checkpoint: checkpoints.NewCheckpointV1Now()}, nil
+}
+
+func inventorySorter(a *extractor.Inventory, b *extractor.Inventory) int {
+	return cmp.Or(cmp.Compare(a.Name, b.Name), cmp.Compare(a.Version, b.Version))
+}
+
+func (di *Diff) scalibrInventory(ctx context.Context, repoURL string, ref string) ([]*extractor.Inventory, error) {
+	clone, err := di.cli.Clone(ctx, repoURL, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	tree, err := clone.Worktree()
+	if err != nil {
+		return nil, err
+	}
+	return scanFs(ctx, tree.Filesystem, map[string]string{})
+}
+
+func scanFs(ctx context.Context, memFS billy.Filesystem, _ map[string]string) ([]*extractor.Inventory, error) {
+	// have to down-cast here, because scalibr needs multiple io/fs types
+	wrapped, ok := iofs.New(memFS).(scalibr_fs.FS)
+	if !ok {
+		return nil, fmt.Errorf("error converting filesystem to ReadDirFS")
+	}
+
+	desiredCaps := scalibr_plugin.Capabilities{
+		OS:            scalibr_plugin.OSAny,
```

Comment:
OSAny is supposed to be used for specifying plugin requirements only. When specifying the scan environment's capabilities you should pick a specific OS.
You can use the SCALIBR-prodivded platform.OS() function to determine the OS at runtime: https://github.com/google/osv-scalibr/blob/main/binary/cli/cli.go#L418